### PR TITLE
chore(*): switch from JSHint/JSCS to ESLint

### PR DESCRIPTION
**Pull request by mgol**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
A build-related refactor.

**What is the current behavior? (You can also link to an open issue here)**
Code is linted via JSHint+JSCS.

**What is the new behavior (if this is a feature change)?**
Code is linted via ESLint.

**Does this PR introduce a breaking change?**
No.

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] ~~Tests for the changes have been added (for bug fixes / features)~~
- [x] ~~Docs have been added / updated (for bug fixes / features)~~

**Other information**:
This is the last commit of PR #14952 backported to the `v1.5.x` branch with some required changes and `.catch -> ['catch']`, `.finally -> ['finally']`. The other commits from that PR are already backported; they weren't dangerous and cherry-picked cleanly.

This needs to be reviewed with two things in mind: the PR shouldn't introduce security issues and backwards incompatible changes.

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Only 308 files to review :-/

**Comment by petebacondarwin** [comment]
I have LGTM the following files:
* parse.js
* sce.js
* compile.js
* sanitize.js
* jqlite.js

File: docs/app/assets/js/search-worker.js [hunk 1]
```diff
@@ -25,13 +26,13 @@ searchDataRequest.onload = function() {
   searchData.forEach(function(page) {
     index.add(page);
   });
-  postMessage({ e: 'index-ready' });
+  postMessage({e: 'index-ready'});
```
Comment (gkalpak) [inline docs/app/assets/js/search-worker.js]: Ooc, couldn't this also be `self.postMessage`? Is there any reason why you usd it below but not here (just being curious)?

File: docs/app/src/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "root": true,
+  "extends": "../../../.eslintrc-browser.json",
+
+  "globals": {
+    // ngMocks
+    "module": false,
+    "inject": true,
```
Comment (gkalpak) [inline docs/app/src/.eslintrc.json]: I don't think we need these two globals (I missed that in master :scream: ).

File: test/ng/animateCssSpec.js [hunk 3]
```diff
@@ -126,7 +126,7 @@ describe("$animateCss", function() {
       var cancelSpy = jasmine.createSpy();
       var runner = $animateCss(element, { addClass: 'cool' }).start();
 
-      runner.catch(cancelSpy);
+      runner['catch'](cancelSpy);
```
Comment (gkalpak) [inline test/ng/animateCssSpec.js]: I don't these changes are necessary in tests, but OK :no_mouth:

**Comment by gkalpak** [comment]
A couple of minor remarks/questions, but nothing blocking.
LGTM as soon as Travis is happy.

File: docs/app/src/.eslintrc.json (hunk 2 above)
Comment (mgol) [inline docs/app/src/.eslintrc.json]: Good point; I'll add it to the list of things to be fixed once this PR lands. ;)

File: docs/app/assets/js/search-worker.js (hunk 1 above)
Comment (mgol) [inline docs/app/assets/js/search-worker.js]: Good point; I'll fix that after this PR lands on `master` and then backport it.

File: test/ng/animateCssSpec.js (hunk 3 above)
Comment (mgol) [inline test/ng/animateCssSpec.js]: Since we try to support Android 2.3 in 1.5, I think it's fine to try to ensure tests at least parse there. ;P

**Comment by Narretz** [comment]
parse.js, sanitize.js, compile.js, jqlite.js and sce.js LGTM

**Comment by gkalpak** [comment]
Merged as b1bc251.

**Pull request closed without merging**